### PR TITLE
Implement tenant selector

### PR DESCRIPTION
**Pull request by nati**
- switched to use sessionStorage instead of cookie

### CODE REVIEW

**Comment by stepek** [comment]
Your commit decrees code coverage 

Before
```
File                  |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
  userModel.js        |    97.85 |      100 |      100 |      100 |                |
```
After
```
File                  |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
  userModel.js        |    70.23 |    78.95 |    85.71 |    53.33 |... 263,264,265 |
```

File: .eslintrc [hunk 1]
```diff
@@ -7,6 +7,11 @@
     "es6": true,
     "node": true
   },
+  "globals": {
```
Comment (stepek) [inline .eslintrc]: Please add /* global $, sessionStorage */ to userModel.js and import Backbone.
Using globals in eslint config is bad.

File: app/js/models/userModel.js [hunk 2]
```diff
@@ -1,5 +1,4 @@
 import {Model} from 'backbone';
```
Comment (stepek) [inline app/js/models/userModel.js]: If you need Backbone variable please use this and do not use globals variables. 
```import Backbone, {Model} from 'backbone';```

File: app/js/models/userModel.js [hunk 3]
```diff
@@ -19,18 +18,19 @@ export default class UserModel extends Model {
...
   }
 
   /**
    * Calls setAuthData function.
    * @override Model.parse
    * @param {Object} data
    */
-  parse(data) {
-    this.setAuthData(data);
+  parse( data ) {
```
Comment (stepek) [inline app/js/models/userModel.js]: Please use code convention from the project and do not add spaces after and before param.

File: app/js/models/userModel.js [hunk 4]
```diff
@@ -1,5 +1,4 @@
 import {Model} from 'backbone';
-import cookie from 'js-cookie';
```
Comment (stepek) [inline app/js/models/userModel.js]: Please remove js-cookie from package.json

File: app/js/models/userModel.js [hunk 5]
```diff
@@ -46,94 +46,224 @@ export default class UserModel extends Model {
     };
     super.sync(method, model, options);
   }
+  /**
+   * Ajax request to the server
+   * @param {options} options
+   */
+  ajax( options ) {
```
Comment (stepek) [inline app/js/models/userModel.js]: Really we need ajax method in prototype of UserModel. i don't think so.

File: app/js/models/userModel.js [hunk 6]
```diff
@@ -46,94 +46,224 @@ export default class UserModel extends Model {
...
       const authData = {
         auth: {
           passwordCredentials: {
             username: id,
             password
+          }
+        }
+      };
+      const tenants = this.tenants();
+      if ( tenants ) {
```
Comment (stepek) [inline app/js/models/userModel.js]: please add empty line before if block, it will be easy to read.

File: app/js/models/userModel.js [hunk 7]
```diff
@@ -46,94 +46,224 @@ export default class UserModel extends Model {
...
-      return '';
+    const tenant = this.tenantName();
+    const scopedToken = this.getItem( 'scopedToken' );
+    if ( scopedToken ) {
+      const data = scopedToken[tenant];
+      if ( data ) {
+        return data.access.token.id;
+      }
+    }
+    return undefined;
```
Comment (stepek) [inline app/js/models/userModel.js]: If you want return ```undefined``` you don't need specify it.

**Comment by thayashi** [comment]
I confirmed this works in fake keystone, but also I want to confirm it works in real keystone env, give me some time to check it.

File: .eslintrc (hunk 1 above)
Comment (nati) [inline .eslintrc]: done

File: app/js/models/userModel.js (hunk 2 above)
Comment (nati) [inline app/js/models/userModel.js]: done

File: app/js/models/userModel.js (hunk 4 above)
Comment (nati) [inline app/js/models/userModel.js]: done

File: app/js/models/userModel.js (hunk 5 above)
Comment (nati) [inline app/js/models/userModel.js]: done

File: app/js/models/userModel.js (hunk 6 above)
Comment (nati) [inline app/js/models/userModel.js]: done

File: app/js/models/userModel.js (hunk 7 above)
Comment (nati) [inline app/js/models/userModel.js]: done

**Comment by nati** [comment]
@thayashi Thanks!
@stepek thank you for your review! Do you know how to mock Backbone with the way to import Backbone as you suggested?

**Comment by nati** [comment]
@stepek nevermind I could figure out how to mock it

**Comment by nati** [comment]
ut fixed.

**Comment by thayashi** [comment]
When I logged in as admin, even I changed tenant, the resource list shows everything, not filtered by tenant. Is this correct behavior?? If admin, may be it doesn't need to show tenant switch ui?

File: app/templates/header.html [hunk 8]
```diff
@@ -14,6 +14,20 @@
     </li>
   </ul>
 </div>
+<div class="dropdown">
```
Comment (thayashi) [inline app/templates/header.html]: if tenants.length == 1, let's just show the tenant name without drop-down link

Comment (nati) [inline app/templates/header.html]: done

**Comment by nati** [comment]
Yes. it is expected behavior for admin.
This is more like coming from server side implementation.

**Comment by thayashi** [comment]
LGTM, I'll tweak template and css later.

**Pull request merged**